subsys/net/ip/tcp2.c

```diff
@@ -1275,6 +1279,110 @@ static enum net_verdict tcp_recv(struct net_conn *net_conn,
 	return NET_DROP;
 }
 
+static uint32_t seq_scale(uint32_t *seq)
+{
+	return *seq + (k_ticks_to_ns_floor32(k_uptime_ticks()) >> 6);
+}
+
+static uint8_t secret_key[16]; /* MD5 128 bits as described in RFC6528 */
+
+static uint32_t tcpv6_init_isn(struct in6_addr *saddr,
+			       struct in6_addr *daddr,
+			       uint16_t sport,
+			       uint16_t dport)
+{
+	struct {
+		uint8_t secret_key[sizeof(secret_key)];
```

Comment:
This looks like a riddle to me. Can we please get different names for global var vs structure field? Note that MISRA C prohibits variable shadowing. I consider that a stupid rule, but cases like this are apparently the reason they've come up with that rule at all.

Similarly, the struct itself could use a comment why it exists in the first place ("Put various fields in a structure, so they can be hashed together as one block of memory").

Reply:
Btw, I hope it was accounted for the fact that structure field alignment in C is implementation-defined (there can be arbitrary gaps between fields). That shouldn't matter here - we're looking for not-easy-to-guess value, not reproducibility across systems. But maybe it's worth an explicit comment too. (Because sooner or later we'll get a report from some static analysis tool about this matter, so forward-looking comment that it's false positive would be helpful.)

Reply:
Is this really shadowing? Field names and types are in completely different namespaces. Maybe one of them could be clearer, but I could also easily see both being perfectly readable names.

Reply:
BTW, an easy way to handle this is to define something like:
```c
#define SECRET_KEY_LEN 16
static uint8_t secret_key[SECRET_KEY_LEN];
```
and then use something similar in the struct field definition.